models/engine.go

```diff
@@ -0,0 +1,81 @@
+package models
+
+import "context"
+
+// Engine defines a processing unit
+type Engine interface {
```

Comment:
Yes, I agree. I think for now is ok like you said. I also think this might be related to the EngineStatus maybe, so if one is about signaling status it can also be about observability